rp2040: dma: transfer_size_bytes is not respected

The field exists in `TransferConfig` but in `trigger_transfer` the `.DATA_SIZE` register is always set to `.SIZE_BYTE`

This is what the fix should look like, I just don't get how to get it to compile (It complains because the config value isn't available at comptime)

```diff
diff --git a/board-support/raspberrypi-rp2040/src/hal/dma.zig b/board-support/raspberrypi-rp2040/src/hal/dma.zig
index fcbc3d8..b5eb749 100644
--- a/board-support/raspberrypi-rp2040/src/hal/dma.zig
+++ b/board-support/raspberrypi-rp2040/src/hal/dma.zig
@@ -107,7 +107,12 @@ pub const Channel = enum(u4) {
         regs.ctrl_trig.modify(.{
             .EN = @intFromBool(config.enable),
             .DATA_SIZE = .{
-                .value = .SIZE_BYTE,
+                .value = switch (config.transfer_size_bytes) {
+                    1 => .SIZE_BYTE,
+                    2 => .SIZE_HALFWORD,
+                    3 => .SIZE_WORD,
+                    else => unreachable,
+                },
             },
             .INCR_READ = @intFromBool(config.read_increment),
             .INCR_WRITE = @intFromBool(config.write_increment),
```

```
src/hal/dma.zig:110:26: error: value with comptime-only type '@TypeOf(.enum_literal)' depends on runtime control flo
  w
                  .value = switch (config.transfer_size_bytes) {
```

--- a/board-support/raspberrypi-rp2040/src/hal/dma.zig
+++ b/board-support/raspberrypi-rp2040/src/hal/dma.zig
@@ -107,7 +107,12 @@ pub const Channel = enum(u4) {
         regs.ctrl_trig.modify(.{
             .EN = @intFromBool(config.enable),
             .DATA_SIZE = .{
-                .value = .SIZE_BYTE,
+                .value = switch (config.transfer_size_bytes) {
+                    1 => @TypeOf(regs.ctrl_trig.read().DATA_SIZE.value).SIZE_BYTE,
+                    2 => .SIZE_HALFWORD,
+                    4 => .SIZE_WORD,
+                    else => unreachable,
+                },
             },
             .INCR_READ = @intFromBool(config.read_increment),
             .INCR_WRITE = @intFromBool(config.write_increment),

